src/components/Auth.jsx

```diff
@@ -62,125 +49,126 @@ const NickName = ({ isValid, setIsValid }) => {
         닉네임
       </label>
       <input
-        className={`auth__input ${
-          !isValid && message !== "" ? "auth__invalid" : ""
-        }`}
-        type="text"
+        className={`auth__input ${isMessageVisible ? "auth__invalid" : ""}`}
+        type="nickname"
+        name="nickname"
         id="nickname"
         placeholder="닉네임을 입력해주세요"
         required
         value={value}
         onBlur={focusoutAuthInput}
-        onChange={({ target }) => setValue(target.value)}
+        onChange={getUserInfo}
       />
-      <span
-        className={isValid ? "hidden" : ""}
-        style={{
-          padding: "8px 16px 0px",
-          color: "#f74747",
-        }}
-      >
-        {message}
-      </span>
+      {isMessageVisible && <span className="auth__message">{message}</span>}
     </div>
   );
 };
 
-const Password = ({ isValid, setIsValid }) => {
-  const [value, setValue] = useState("");
-  const [message, setMessage] = useState("");
+const Password = ({ value, message, getUserInfo }) => {
+  const [isMessageVisible, setIsMessageVisible] = useState(false);
+  const [type, setType] = useState("password");
 
-  const focusoutAuthInput = ({ target }) => {
-    setMessage(validPassword(target));
-    setIsValid(validPassword(target) === "" && value !== "");
+  const focusoutAuthInput = () => {
+    setIsMessageVisible(message !== "");
   };
 
-  const keyupCheckPasswordSame = ({ target }) => {};
+  const handleValueVisible = () => {
+    setType((prev) => {
+      if (prev === "password") return "text";
+      return "password";
+    });
+  };
 
   return (
     <div className="auth__el">
-      <button className="auth__toggle-btn" type="button">
-        <img src={btnOffImg} alt="visibility off" width="24" height="24" />
-      </button>
-      <button
-        className="auth__toggle-btn auth__toggle-btn--visible"
-        type="button"
-      >
-        <img src={btnOnImg} alt="visibility on" width="24" height="24" />
-      </button>
+      {type === "password" && (
+        <button
+          className="auth__toggle-btn"
+          type="button"
+          onClick={handleValueVisible}
+        >
+          <img src={btnOffImg} alt="visibility off" width="24" height="24" />
+        </button>
+      )}
+      {type === "text" && (
+        <button
+          className="auth__toggle-btn"
+          type="button"
+          onClick={handleValueVisible}
+        >
+          <img src={btnOnImg} alt="visibility on" width="24" height="24" />
+        </button>
+      )}
```

Comment:
```suggestion
const Password = ({ value, message, getUserInfo }) => {
const [isPasswordVisible, setIsPasswordVisible] = useState(false)
return {

        <button
          className="auth__toggle-btn"
          type="button"
          onClick={handleValueVisible}
        >
          <img
          src={isPasswordVisible ? btnOnImg : btnOffImg}
          alt={isPasswordVisible ? "visibility on" : "visibility off"}
          width="24"
          height="24"
          />
        </button>

}
```
 props에서 직접 계산하면바껴야 하는 부분만 콕 찝어서 관리하는것도 가능합니다.